### ISSUE TITLE
Changing partition for /home and /boot deletes the entire installation

### ISSUE THREAD

**Issue by ShadowOfTheDark669** [linuxmint/timeshift#362]
**Bug**
In the process of restoring a snapshot, if the `root` partition(empty) is MANUALLY selected for the `/home` and `/boot` directories instead of the `Keep on root device` option, then timeshift indeed restores the snapshot but deletes all the restored files followed by a `Snapshot restored with errors` popup. It attempts to mount `/home` and `/boot` both at `/` and deletes everything to do so, causing errors and an unrecoverable system. Checking with a file manager upon restore also ensures that the entire installation is deleted barring a few EMPTY folders.

**To Reproduce**
1. Select a snapshot and click restore
2. Select the root partition, then for the `/home` and `/boot` directories, which have `Keep on root device` by default, select the desired partition for `root` manually. Continue with the rest of the menus normally.
3. Timeshift restores as expected but at the end of restoring, no of restored files is matched/exceeded to the no of deleted files.
4. `Snapshot restored with errors` popup appears.

**Expected behavior**
Timeshift restores the snapshots normally.
**Screenshots**

![Screenshot_20241101_171157](https://github.com/user-attachments/assets/98d5000c-dbd5-4941-9d35-d1a86f774d9e)
![Screenshot_20241101_171414](https://github.com/user-attachments/assets/936da48a-c9c6-47e7-b120-bf4fbbaa6418)
![Screenshot_20241101_171812](https://github.com/user-attachments/assets/ab8ae284-9f1d-45c8-80ac-752e720c9bd2)

**System:**
 - Linux Distribution Name and Version: OpenSUSE Leap 15.6 , Fedora 40 and Void Linux Glibc
 - Desktop: KDE 6.2 , GNOME 46 , XFCE idk version
 - Application Version: 24.06.3 on Void Linux although issue appears on all the three distros.

**Comments**
Timeshift should atleast warn at the selection screen of this behaviour, or change the default behaviour to mount `/boot` and `/home` at `/` for the afore mentioned configuration instead of deleting everything. The only extremely subtle hint is at the second last conformation screen before restore.

**Comment by zahlman** [comment]
When you select a partition for the `/home` and `/boot` folders, you are asking for them to be restored *at the root of* the selected partition. "Keep on root partition" is a special case - the data goes to a subfolder on the partition where `/` is restored, i.e. is *not* configured for separate mounting.

Generally the UI should warn about, or prevent, restoring more than one of {`/`, `/boot`, `/home`} to the same partition, yes. I don't think it makes sense to do special cases instead - a user who wants `/home` data on the `/` partition can just re-select "Keep on root device".

**Comment by ShadowOfTheDark669** [comment]
> When you select a partition for the `/home` and `/boot` folders, you are asking for them to be restored _at the root of_ the selected partition. "Keep on root partition" is a special case - the data goes to a subfolder on the partition where `/` is restored, i.e. is _not_ configured for separate mounting.
> 
> Generally the UI should warn about, or prevent, restoring more than one of {`/`, `/boot`, `/home`} to the same partition, yes. I don't think it makes sense to do special cases instead - a user who wants `/home` data on the `/` partition can just re-select "Keep on root device".

Yep, I think warning or preventing such restore is better than the special case which I expressed in my post.
